### ISSUE TITLE
Is Timmy Okay?

### ISSUE THREAD

**Issue by Schwad** [scarpe-team/scarpe#457]
It appears at some point the layouts for timmy got a bit bonkers. 

<img width="998" alt="image" src="https://github.com/scarpe-team/scarpe/assets/7865030/ad9fa49c-e1b7-4463-9e6a-ceca95dc65f8">

Mayhaps mostly to do with respecting flow/stack height/width. I'll see if I can bisect

Unrelated but it looks damn cool with tiranti

<img width="1001" alt="image" src="https://github.com/scarpe-team/scarpe/assets/7865030/01384a7e-ab6a-4f2f-9e10-ebb585b35485">

**Comment by Schwad** [comment]
I just ran a bisect, here's the commit where the alignment went: 9a134cccd7556fbca642d5b23c6c25539b5e3f48

**Comment by noahgibbs** [comment]
Oh, interesting. Yeah, I'll look into that. We *do* want an outer div wrapper, but maybe that was the wrong way to do it...

**Comment by Schwad** [comment]
Here's a snippet of the current HTML from debug

```
<div id='2-wrap'>
  <div id="2" style="display:flex;flex-direction:row;flex-wrap:wrap;align-content:flex-start;justify-content:flex-start;align-items:flex-start;background:black;width:100%">
    <p align="center" id="3" style="color:#FF0000;font-size:48px">You've downloaded a virus!</p>
    
    <div id='4-wrap'>
      <div id="4" style="display:flex;flex-direction:row;flex-wrap:wrap;align-content:flex-start;justify-content:flex-start;align-items:flex-start;width:100%;height:30%">
        
        <div id='5-wrap'>
          <div id="5" style="display:flex;flex-direction:column;align-content:flex-start;justify-content:flex-start;align-items:flex-start;background:white;width:100%">
            <p align="center" id="6" style="font-size:34px">You thought this was a recipe for cookies, but it was me, your arch nemesis!<br><br></p>
            
            <p align="center" id="7" style="font-size:30px">I have taken over your computer and will not give it back until you pay me $1,000,000.</p>
          </div>
        </div>

      </div>
    </div>

    <!-- ... Other similar divs ... -->

  </div>
</div>

<div id="root-fonts"></div>
<div id="root-alerts"></div>
```

**Comment by Schwad** [comment]
and prior, just taken from like september or august 

<div id="3020" style="display:flex;flex-direction:row;flex-wrap:wrap;align-content:flex-start;justify-content:flex-start;align-items:flex-start;background:black;width:100%;height:100%">
  <p align="center" id="2620" style="color:#FF0000;font-size:48px">You've downloaded a virus!</p>
  
  <div id="2700" style="display:flex;flex-direction:row;flex-wrap:wrap;align-content:flex-start;justify-content:flex-start;align-items:flex-start;width:100%;height:30%">
    
    <div id="2680" style="display:flex;flex-direction:column;align-content:flex-start;justify-content:flex-start;align-items:flex-start;background:white;width:100%">
      <p align="center" id="2640" style="font-size:34px">You thought this was a recipe for cookies, but it was me, your arch nemesis!<br><br></p>
      
      <p align="center" id="2660" style="font-size:30px">I have taken over your computer and will not give it back until you pay me $1,000,000.</p>
    </div>

  </div>

  
  
</div>

**Comment by Schwad** [comment]
> Oh, interesting. Yeah, I'll look into that. We do want an outer div wrapper, but maybe that was the wrong way to do it...

@noahgibbs  Honestly no worries and I'm happy to take a peek myself too at some point. Just trying to get in the good hygiene of filing issues if I have a thought or see something.

Going to a meetup tonight so was looking through some of the examples :)

**Comment by noahgibbs** [comment]
That's a great idea -- definitely no problem. The wrapper div was to ensure good cleanup - if we *know* there's one specific outer element, it's easy to guarantee we've cleaned everything. But there are other ways to make that idea work without a no-formatting outer div.

I'm used to thinking of "wrap a div with no attributes around the outside" as a harmless no-op. But as this demonstrates, that's not always true.

**Comment by Schwad** [comment]
is the wrapper div set to be width and height 100%? in theory wouldn't the inner divs for flow and stack respect that with their widths and height?

**Comment by Schwad** [comment]
If not I wonder if that could help. but again I'm mentally operating with less context on the change than yourself 😅

**Comment by noahgibbs** [comment]
The wrapper div has no attributes of any kind - just an ID.

**Comment by noahgibbs** [comment]
If I make minimal changes to delete the wrapper div, I get the old Timmy look back. I need to check Timmy and see if it makes sense how it's rendering... But you're right, the div changes it.

This also made me look back and realise I was doing the html_wrapper_element method wrong. Here are the changes I made to fix that problem and remove the wrapper div:

~~~
diff --git a/lib/scarpe/wv/drawable.rb b/lib/scarpe/wv/drawable.rb
index 565f993..d0fb9a4 100644
--- a/lib/scarpe/wv/drawable.rb
+++ b/lib/scarpe/wv/drawable.rb
@@ -197,7 +197,8 @@ module Scarpe::Webview
     #
     # @return [Scarpe::WebWrangler::ElementWrangler] a DOM object manager
     def html_wrapper_element
-      @elt_wrangler ||= Scarpe::Webview::WebWrangler::ElementWrangler.new(html_id)
+      #@wrapper_elt_wrangler ||= Scarpe::Webview::WebWrangler::ElementWrangler.new("#{html_id}-wrap")
+      html_element
     end

     # Return a promise that guarantees all currently-requested changes have completed
@@ -223,7 +224,8 @@ module Scarpe::Webview
     def to_html
       @children ||= []
       child_markup = @children.map(&:to_html).join
-      "<div id='#{html_id}-wrap'>" + element { child_markup } + "</div>"
+      #"<div id='#{html_id}-wrap'>" + element { child_markup } + "</div>"
+      element { child_markup }
     end

     # This binds a Scarpe JS callback, handled via a single dispatch point in the app
~~~

I'm going to try to make a change that keeps what I wanted from that diff. But this would be enough to return Timmy to how it used to render (at least for me.)

**Comment by noahgibbs** [comment]
Fixed by #459.
